### PR TITLE
Update version of `rcgen` we use

### PR DESCRIPTION
**Pull request by ionut-arm**
Catching up to date with the latest release of `rcgen`.

Also fixing a clippy lint issue.

Signed-off-by: Ionut Mihalcea <ionut.mihalcea@arm.com>

### CODE REVIEW

**Review by anta5010 — APPROVED** [review]

File: src/subcommands/decrypt.rs [hunk 1]
```diff
@@ -24,7 +24,7 @@ pub struct Decrypt {
 impl Decrypt {
     /// Decrypts data.
     pub fn run(&self, basic_client: BasicClient) -> Result<()> {
-        let input = base64::decode(self.input_data.as_bytes().to_vec())?;
```
Comment (paulhowardarm) [inline src/subcommands/decrypt.rs:27]: Just curious as to what prompted this change? We're not consuming rcgen at this point. Is that just a bit of unrelated clean-up?
